File: gymflow-frontend/src/services/receptionistService.js
import apiClient from '../apiClient';
import { RECEPTIONIST_ENDPOINTS } from '../apiConfig';

export const getReceptionists = async () => {
  try {
    const response = await apiClient.get(RECEPTIONIST_ENDPOINTS.GET_ALL);
    return response.data;
  } catch (error) {
    console.error("Error fetching receptionists", error);
    throw error;
  }
};

export const getReceptionistById = async (id) => {
  const { data } = await apiClient.get(RECEPTIONIST_ENDPOINTS.GET_BY_ID(id));
  return data;
};

export const createReceptionist = async (payload) => {
  const { data } = await apiClient.post(RECEPTIONIST_ENDPOINTS.CREATE, payload);
  return data;
};

export const updateReceptionist = async (id, payload) => {
  const { data } = await apiClient.put(RECEPTIONIST_ENDPOINTS.UPDATE(id), payload);
  return data;
};

export const deleteReceptionist = async (id) => {
  const { data } = await apiClient.delete(RECEPTIONIST_ENDPOINTS.DELETE(id));
  return data;
};

export const activateReceptionist = async (id) => {
  const { data } = await apiClient.put(RECEPTIONIST_ENDPOINTS.ACTIVATE(id));
  return data;
};

export const deactivateReceptionist = async (id) => {
  const { data } = await apiClient.put(RECEPTIONIST_ENDPOINTS.DEACTIVATE(id));
  return data;
};

File: gymflow/src/main/java/com/application/gymflow/service/ReceptionistService.java
package com.application.gymflow.service;



import com.application.gymflow.dto.receptionist.ReceptionistCreateRequestDto;
import com.application.gymflow.dto.receptionist.ReceptionistResponseDto;

import java.util.List;


public interface ReceptionistService {

    /**
     * Create a new Receptionist (with linked User account).
     *
     * @param request details for creating Receptionist & User
     * @return created ReceptionistResponseDto
     */
    ReceptionistResponseDto createReceptionist(ReceptionistCreateRequestDto request);

    /**
     * Update an existing Receptionist by ID.
     *
     * @param id      Receptionist Long
     * @param request updated details
     * @return updated ReceptionistResponseDto
     */
    ReceptionistResponseDto updateReceptionist(Long id, ReceptionistCreateRequestDto request);

    /**
     * Get a Receptionist by ID.
     *
     * @param id Receptionist Long
     * @return ReceptionistResponseDto
     */
    ReceptionistResponseDto getReceptionistById(Long id);

    /**
     * Get all Receptionists.
     *
     * @return List of ReceptionistResponseDto
     */
    List<ReceptionistResponseDto> getAllReceptionists();

    /**
     * Soft delete a Receptionist (disable User + Receptionist).
     *
     * @param id Receptionist Long
     */
    void deleteReceptionist(Long id);

    /**
     * Activate a Receptionist (enable User + mark active).
     *
     * @param id Receptionist Long
     * @return activated ReceptionistResponseDto
     */
    ReceptionistResponseDto activateReceptionist(Long id);

    /**
     * Deactivate a Receptionist (disable User + mark inactive).
     *
     * @param id Receptionist Long
     * @return deactivated ReceptionistResponseDto
     */
    ReceptionistResponseDto deactivateReceptionist(Long id);
}
